lib/resque/plugins/retry.rb

```diff
@@ -453,6 +453,13 @@ def on_failure_retry(exception, *args)
           return
         end
 
+        # If we are "ignoring" the exception, then we decrement the retry
+        # counter, so that the current attempt didn't count toward the retry
+        # counter.
+        if !@ignore_exceptions.nil? && @ignore_exceptions.include?(exception.class)
```

Comment:
Do you think it might make more sense to move the ignore logic up to the code-block that it incrementing the counter? Rather than incrementing then decrementing here?